AutoCollection/diagnostic-channel/SpanParser.ts

```diff
@@ -17,22 +17,19 @@ export function spanToTelemetryContract(span: Span): (Contracts.DependencyTeleme
     const id = `|${span.spanContext().traceId}.${span.spanContext().spanId}.`;
     const duration = Math.round(span["_duration"][0] * 1e3 + span["_duration"][1] / 1e6);
     let peerAddress = span.attributes["peer.address"] ? span.attributes["peer.address"].toString() : "";
-    let component = span.attributes["component"] ? span.attributes["component"].toString() : "";
 
-    const isHttp: boolean = ((component).toUpperCase() === Constants.DependencyTypeName.Http) || (!!span.attributes[Constants.SpanAttribute.HttpUrl]);
-    const isGrpc: boolean = (component).toLowerCase() === Constants.DependencyTypeName.Grpc;
+    const isHttp: boolean = (!!span.attributes[Constants.SpanAttribute.HttpStatusCode]) || (!!span.attributes[Constants.SpanAttribute.HttpUrl]);
+    const isGrpc: boolean = (!!span.attributes[Constants.SpanAttribute.GrpcStatusCode]);
```

Comment:
- to check if it's http, check span[HTTP_STATUS_CODE] [reference](https://github.com/Azure/azure-sdk-for-js/blob/df4ad49a92134f94540b43ed34bbe5605eb1164a/sdk/monitor/monitor-opentelemetry-exporter/src/utils/spanUtils.ts#L110)
- to check if it's grpc, check span[RPC_GRPC_STATUS_CODE] [reference](https://github.com/Azure/azure-sdk-for-js/blob/df4ad49a92134f94540b43ed34bbe5605eb1164a/sdk/monitor/monitor-opentelemetry-exporter/src/utils/spanUtils.ts#L115)